### PR TITLE
[release/8.0] Add browser security headers to dashboard

### PR DESCRIPTION
**Pull request by JamesNK**
## Customer Impact

Part of security work to include recommended headers.

## Testing

Manual and unit tests

## Risk

Low

## Regression?

No
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3511)

### CODE REVIEW

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
@JamesNK once the required approvals are acquired this will get merged.

**Review by tlmii — APPROVED** [review]

**Comment by joperezr** [comment]
[For tracking] This was a backport of #3354

**Pull request merged**
